### ISSUE TITLE
Have full control of unicode input chars, like invisible spaces, and glaemscribe mode language specific characters 

### ISSUE THREAD

**Issue by BenTalagan** [BenTalagan/glaemscribe#18]
This discussion was originally started in #15.

Glaemscribe did not give the possibility to have certain chars as input for a mode, either because they are invisible chars (word joiners, & so on) so they are extremely hard to write in the mode file, or because they are part of the language (`* [ ] ( ) _ ,` and so on).

This is now fixed by commit 06c4fea6558363e290e7d45efff8c9bacf59ffda with the new following feature, unicode vars can be used in rule source members and also in variable definitions. The syntax is {UNI_XXXX} with XXXX being a hexadecimal value.

The following shortcuts have been added to the engine : 

```ruby
        # Characters that are not easily entered or visible in a text editor
        add_var("NBSP",           "{UNI_A0}")
        add_var("WJ",             "{UNI_2060}")
        add_var("ZWSP",           "{UNI_200B}")
        add_var("ZWNJ",           "{UNI_200C}")        

        # The following characters are used by the mode syntax.
        add_var("UNDERSCORE",     "{UNI_5F}")
        add_var("ASTERISK",       "{UNI_2A}")
        add_var("COMMA",          "{UNI_2C}")
        add_var("LPAREN",         "{UNI_28}")
        add_var("RPAREN",         "{UNI_29}")
        add_var("LBRACKET",       "{UNI_5B}")
        add_var("RBRACKET",       "{UNI_5D}")
```

Will be released with the branch TolkiensReadingDay2018 .

**Comment by laicasaane** [comment]
Hi, I can't figure out how to use this syntax in “variable definitions”. Could you please give some examples?

Am I wrong to expect the engine would produce any unicode character upon input, bypass the charset definition? Like, I write {UNI_61} and eventually there will be an “a” character in the transcribed text?

**Comment by BenTalagan** [comment]
Yes, any {UNI_HEX} will stand for the character in HEX. One example : 

```
{MY_A} === {UNI_61}
{MY_A} --> YANTA
```

Will produce a rule for 'a' transcribed to the yanta tengwa.

You can also use the syntax in **left** members of a rule (not right members, since it's not homogenous with the output) :

 ```
{UNI_61}bc --> YANTA
```
will create a rule that transcribes the sequence 'abc' to yanta.

**Comment by BenTalagan** [comment]
Closed after 1.2.0 release.
